cherry_picker/cherry_picker.py

```diff
@@ -843,23 +849,33 @@ def validate_sha(sha):
         )
 
 
-def version_from_branch(branch):
+def version_sort_key(config, branch):
     """
-    return version information from a git branch name
+    Get sort key based on version information from the given git branch name.
+
+    This function can be used as a sort key in list.sort()/sorted() provided that
+    you additionally pass config as a first argument by e.g. wrapping it with
+    functools.partial().
+
+    Branches with version information come first and are sorted from latest
+    to oldest version.
+    Branches without version information come second and are sorted alphabetically.
     """
     try:
-        return tuple(
-            map(
-                int,
-                re.match(r"^.*(?P<version>\d+(\.\d+)+).*$", branch)
-                .groupdict()["version"]
-                .split("."),
-            )
-        )
+        match = re.match(r"^.*(?P<version>\d+(\.\d+)+).*$", branch)
+        raw_version = match.groupdict()["version"].split(".")
     except AttributeError as attr_err:
-        raise ValueError(
-            f"Branch {branch} seems to not have a version in its name."
-        ) from attr_err
+        if not branch:
+            raise ValueError("Branch name is an empty string.") from attr_err
+        if config["require_version_in_branch_name"]:
+            raise ValueError(
+                f"Branch {branch} seems to not have a version in its name."
+            ) from attr_err
+        # Use '0' to sort regular branch names *after* version numbers
```

Comment:
```suggestion
        # Use 1 to sort regular branch names *after* version numbers
```
Typo, and the quotes can be dropped since it's not a string (anymore?).

Reply:
I probably just quoted it to indicate it's an exact value that I'm quoting from the next line, I don't really think this was ever a string. But I guess that doesn't matter, not using quotes seems less confusing anyway 👍